shiksha-api/app-service/app/services/question_paper_service.py

```
@@ -328,209 +198,93 @@ def _format_system_prompt(
-                    # rag-adapter does not support structured output, so we pass model json schema for now.
-                    curr_message=user_message + "\n\nResponse format must conform to JSON schema:\n" + json.dumps(GeneratedQuestionItemResponse.model_json_schema()),
-                    chat_history=chat_history
-                )
-                # chat_with_index returns {"response": str, "source_nodes": list}
-                response_content = response_content["response"]
-                content = response_content.strip("```json").strip("```")
-                return GeneratedQuestionItemResponse.model_validate_json(content).items
+                response_content = await rag_adapter.chat_with_index(curr_message=user_message, chat_history=chat_history, output_cls=response_format)
+                items = response_format.model_validate(response_content["response"])
         except Exception as e:
             logger.exception(e)
```

Ruff (0.15.17)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)